vault/identity_store_aliases.go

```diff
@@ -183,14 +208,20 @@ func (i *IdentityStore) handleAliasUpdateCommon() framework.OperationFunc {
 				return nil, fmt.Errorf("existing alias is not associated with an entity")
 			}
 			if canonicalID == "" || entity.ID == canonicalID {
-				// Nothing to do
+				// Update the alias in the entity.
```

Comment:
This is not the right place to update the entity. The entity updated here will not be persisted and hence won't survive a seal/unseal cycle. Instead, what you could do here is just this.
```
if canonicalID == "" {
	canonicalID = entity.ID
}
```

There is an `if` block below that takes care of updating the alias data and the flow continues to persist the entity. On line 245 below, we could do this:
```
                switch {
			case entity == nil:
				// If entity is nil, we didn't find a previous alias from factors,
				// so append to this entity
				entity = canonicalEntity
				entity.Aliases = append(entity.Aliases, alias)
			case entity.ID != canonicalEntity.ID:
				// In this case we found an entity from alias factors but it's not
				// the same, so it's a migration
				previousEntity = entity
				entity = canonicalEntity

				for aliasIndex, item := range previousEntity.Aliases {
					if item.ID == alias.ID {
						previousEntity.Aliases = append(previousEntity.Aliases[:aliasIndex], previousEntity.Aliases[aliasIndex+1:]...)
						break
					}
				}

				entity.Aliases = append(entity.Aliases, alias)
				resp.AddWarning(fmt.Sprintf("alias is being transferred from entity %q to %q", previousEntity.ID, entity.ID))
			default:
				// Update the alias
				for index, item := range entity.Aliases {
					if item.ID == alias.ID {
						entity.Aliases[index] = alias
					}
				}
			}
```

Reply:
@vishalnayak thank you. I have made the changes. This now means that an update operation now returns:

```
Key             Value
---             -----
canonical_id    c40d54ae-4f76-abc1-170b-16b51b95bc1a
id              0a427dc1-1f19-68ee-16e3-7577af67df8b
```

instead of the generic

```
Success! Data written to: identity/entity-alias/id/46cefc30-705e-e40b-0417-324c0b878dd2
```

Reply:
That's true. However, it is possible to distinguish between an alias creation operation vs an alias update operation. In line 141, you might want to set a boolean, say `newAlias` to `true` and use that to respond a `return nil, nil` instead of a `return resp, nil` at the end of the function.

Reply:
I have implemented that change.

Reply:
Hmm, I don't see it though. Did you push it?